pydra/engine/submitter.py

```diff
@@ -142,7 +142,10 @@ def __exit__(self, type, value, traceback):
             if futures:
                 # wait until all states complete or error
                 await asyncio.gather(*futures)
-            return runnable
+            if runnable.state:
+                return runnable
+            else:
+                return job  # dj: not sure why I can't return runnable!!
```

Comment:
this may have to do with states.

Reply:
I might be wrong, but I thing parts for `runnable` with state and without are called in completely different places